generatorcore/industry2018/co2e.py

```diff
@@ -1,15 +1,18 @@
+# pyright: strict
 from dataclasses import dataclass
-from ..utils import element_wise_plus
 
 
 @dataclass
 class CO2e:
     # Used by i, p, p_miner, p_chem, p_metal, p_other
-    CO2e_combustion_based: float = None  # type: ignore
-    CO2e_production_based: float = None  # type: ignore
-    CO2e_total: float = None  # type: ignore
-    energy: float = None  # type: ignore
-    prod_volume: float = None  # type: ignore
+    CO2e_combustion_based: float = 0
```

Comment:
I mildly prefer to only set default values for member variables if they get computed by __post_init__ from other fields. That way one cannot accidentally forget to initialize one of the variables at the call sites and also at the callsites it is more in the readers face that e.g. there is no combustion based CO2e for that callsite.

This of course isn't always true / is a trade off that has to be looked at together with other concerns (for example I ended up deciding against it for the Transport.demand_* fields, to make the call sites more conscise). 

Anyhow I wanted to let you know my thoughts feel free to decide yourself what to do with them.

Reply:
@bgrundmann: Ok verstehe und da stimme ich dir zu.

Wenn ich das allerdings gar nicht initialisiere, dann kriege ich aktuell Probleme, so wie ich es gerade mit der Vererbung und dem post_init implementiert habe.

Oder meintest du, ich solle `= None  # type: ignore stehen lassen?

Reply:
Wahrscheinlich redest du von dem Problem das der von `dataclass` erzeugte Code, dann eine Fehlermeldung erzeugt weil position arguments nach keyword argument erzeugt werden. Das kannst du umgehen indem du keyword arguments erzwingst (was sowieso ne gute idee ist weil alle unsere argumente floats sind und man die deshalb sonst leicht verwechselt).

Siehe https://github.com/GermanZero-de/localzero-generator-core/blob/dc3e03f6ed9df3213f3d9361eb498660ab289efe/generatorcore/transport2030/transport.py#L22

Reply:
Done